### PR TITLE
fix(editor): invalid caret in note-edgeless-block on focus

### PR DESCRIPTION
**Pull request by congzhou09**
### Problem
●In edgeless mode, when starting to edit, `note-block` exhibits two types of invalid caret behavior:
(1)**Title Region Misalignment**: Clicking on the title region incorrectly generates the caret in the first line of the note content, rather than in the title itself.
(2)**Vanishing Caret at Line End**: When clicking in the empty space beyond the end of a text section, the caret appears momentarily at the line's end but disappears immediately.
●The following video demonstrates these issues:

https://github.com/user-attachments/assets/db9c2c50-709f-4d32-912c-0f01841d2024

### Solution
●**Title Click Interception**: Added a check to determine if the click coordinates fall in the title region. If so, the caret positioning is now handled by a dedicated logic path. Otherwise, it falls back to the existing note-content logic as before.
●**Range Normalization**: When the generated `range.startContainer` is not a `TextNode`, try to find a most appropriate `TextNode` and update the `range` accordingly.

### After
●The video below shows the behavior after this fix.

https://github.com/user-attachments/assets/b2f70b64-1fc6-4049-8379-8bcf3a488a05

## Summary by CodeRabbit

* **Bug Fixes**
  * Clicking a page block title no longer creates unwanted paragraphs and reliably focuses the title.
  * Paragraph creation now occurs only when needed and focus is applied only after successful creation.
  * Click coordinates are clamped to container bounds to prevent misplaced cursors or focus.

* **Improvements**
  * Caret normalization: clicks place the caret at the last meaningful text position for consistent single-cursor behavior.

* **Tests**
  * Added end-to-end coverage for caret placement and focus transitions.
  * New ratio-based click/double-click test utilities and a helper for double-clicking note bodies.

### CODE REVIEW

File: blocksuite/affine/shared/src/utils/dom/point-to-range.ts [hunk 1]
```diff
@@ -90,9 +90,59 @@ export function getCurrentNativeRange(selection = window.getSelection()) {
...
+  }
+
+  // Fallback, try to find text in startContainer
+  const text = lastMeaningfulTextNode(startContainer);
+  if (text) {
+    range.setStart(text, text.textContent?.length ?? 0);
+    range.collapse(true);
+    return;
+  }
+}
```
[the remark this answers is not in the document]
Comment (congzhou09) [reply blocksuite/affine/shared/src/utils/dom/point-to-range.ts:160]: "🐛 Proposed fix to handle startOffset edge case" is added.

**Comment by darkskygit** [comment]
can we add a unit test to verify this behavior?

**Comment by congzhou09** [comment]
> can we add a unit test to verify this behavior?

ok, I'll do this later.

**Comment by congzhou09** [comment]
> can we add a unit test to verify this behavior?

Hello, I've added a corresponding test case in the new commit. @darkskygit

**Comment by congzhou09** [comment]
New commit: Fix e2e test cases in this recent [action](https://github.com/toeverything/AFFiNE/actions/runs/21698871442). @darkskygit

**Comment by congzhou09** [comment]
New commit: added unit-test for point-to-range. @darkskygit

**Review by darkskygit — APPROVED** [review]
